# First Steps with Python and Jupyter

![](https://i.imgur.com/gvSnw4A.png)




This Notebook covers the following topics:

* Performing arithmetic operations using Python
* Solving multi-step problems using variables
* Evaluating conditions using Python
* Combining conditions with logical operators
* Adding text styles using Markdown

#### Option 1: Running on your computer locally

To run the code on your computer locally, you'll need to set up [Python](https://www.python.org), download the notebook and install the required libraries. I recommend using the [Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) distribution of Python.

## Performing Arithmetic Operations using Python

Let's begin by using Python as a calculator. You can write and execute Python using a code cell within Jupyter.

> **Working with cells**: To create a new cell within Jupyter, you can select "Insert > Insert Cell Below" from the menu bar or just press the "+" button on the toolbar. You can also use the keyboard shortcut `Esc+B` to create a new cell. Once a cell is created, click on it to select it. You can then change the cell type to code or markdown (text) using the "Cell > Cell Type" menu option. You can also use the keyboard shortcuts `Esc+Y` and `Esc+M`. Double-click a cell to edit the content within the cell. To apply your changes and run a cell, use the "Cell > Run Cells" menu option or click the "Run" button on the toolbar or just use the keyboard shortcut `Shift+Enter`. You can see a full list of keyboard shortcuts using the "Help > Keyboard Shortcuts" menu option.

Run the code cells below to perform calculations and view their result. Try changing the numbers and run the modified cells again to see updated results. Can you guess what the `//`, `%`, and `**` operators are used for?

In [1]:
2 + 3 + 9

14

In [2]:
99 - 73

26

Just like arithmetic operations, the result of a comparison operation can also be stored in a variable.

In [3]:
100 / 7

14.285714285714286

In [4]:
100 // 7

14

In [5]:
100 % 7

2

In [6]:
5 ** 3

125

As you might expect, operators like `/` and `*` take precedence over other operators like `+` and `-` as per mathematical conventions. You can use parentheses, i.e. `(` and `)`, to specify the order in which operations are performed.

In [7]:
((2 + 5) * (17 - 3)) / (4 ** 3)

1.53125

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |


> **`print`**: The `print` function is used to display information. It takes one or more inputs, which can be text (within quotes, e.g., `"this is some text"`), numbers, variables, mathematical expressions, etc. We'll learn more about variables & functions in the next tutorial.

Creating a code cell for each variable or mathematical operation can get tedious. Fortunately, Jupyter allows you to write multiple lines of code within a single code cell. The result of the last line of code within the cell is displayed as the output.

Let's rewrite the solution to our word problem within a single cell.

In [18]:
# Store input data in variables
cost_of_ice_bag = 1.25
profit_margin = .2
number_of_bags = 500

# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = number_of_bags * profit_per_bag

# Display the result
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


Note that we're using the `#` character to add *comments* within our code.

> **Comments**: Comments and blank lines are ignored during execution, but they are useful for providing information to humans (including yourself) about what the code does. Comments can be inline (at the end of some code), on a separate line, or even span multiple lines.

Inline and single-line comments start with `#`, whereas multi-line comments begin and end with three quotes, i.e. `"""`. Here are some examples of code comments:

In [19]:
my_favorite_number = 1 # an inline comment

In [20]:
# This comment gets its own line
my_least_favorite_number = 3

In [21]:
"""This is a multi-line comment.
Write as little or as much as you'd like.

Comments are really helpful for people reading
your code, but try to keep them short & to-the-point.

Also, if you use good variable names, then your code is
often self explanatory, and you may not even need comments!
"""
a_neutral_number = 5



> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [22]:
my_favorite_number = 1
my_least_favorite_number = 5
a_neutral_number = 3

In [23]:
# Equality check - True
my_favorite_number == 1

True

In [24]:
# Equality check - False
my_favorite_number == my_least_favorite_number

False

In [25]:
# Not equal check - True
my_favorite_number != a_neutral_number

True

In [26]:
# Not equal check - False
a_neutral_number != 3

False

In [27]:
# Greater than check - True
my_least_favorite_number > a_neutral_number

True

In [28]:
# Greater than check - False
my_favorite_number > my_least_favorite_number

False

In [29]:
# Less than check - True
my_favorite_number < 10

True

In [30]:
# Less than check - False
my_least_favorite_number < my_favorite_number

False

In [31]:
# Greater than or equal check - True
my_favorite_number >= 1

True

In [32]:
# Greater than or equal check - False
my_favorite_number >= 3

False

In [33]:
# Less than or equal check - True
3 + 6 <= 9

True

In [34]:
# Less than or equal check - False
my_favorite_number + a_neutral_number <= 3

False

In [35]:
cost_of_ice_bag = 1.25
is_ice_bag_expensive = cost_of_ice_bag >= 10
print("Is the ice bag expensive?", is_ice_bag_expensive)

Is the ice bag expensive? False


## Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |


In [36]:
my_favorite_number

1

In [37]:
my_favorite_number > 0 and my_favorite_number <= 3

True

In [38]:
my_favorite_number < 0 and my_favorite_number <= 3

False

In [39]:
my_favorite_number > 0 and my_favorite_number >= 3

False

In [40]:
True and False

False

In [41]:
True and True

True

The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |



In [42]:
a_neutral_number = 3

In [43]:
a_neutral_number == 3 or my_favorite_number < 0

True

In [44]:
a_neutral_number != 3 or my_favorite_number < 0

False

In [45]:
my_favorite_number < 0 or True

True

In [46]:
False or False

False

The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

In [47]:
not a_neutral_number == 3

False

In [48]:
not my_favorite_number < 0

True

In [49]:
not False

True

In [50]:
not True

False

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

In [51]:
(2 > 3 and 4 <= 5) or not (my_favorite_number < 0 and True)

True

In [52]:
not (True and 0 < 1) or (False and True)

False

If parentheses are not used, logical operators are applied from left to right.

In [53]:
not True and 0 < 1 or False and True

False

Experiment with arithmetic, conditional and logical operators in Python using the interactive nature of Jupyter notebooks. We will learn more about variables and functions in future tutorials.

There is also a ternary operator in Python. Learn about it here: https://data-flair.training/blogs/python-ternary-operator/

## Further Reading and References

Following are some resources where you can learn about more arithmetic, conditional and logical operations in Python:

* Python Tutorial at W3Schools: https://www.w3schools.com/python/
* Practical Python Programming: https://dabeaz-course.github.io/practical-python/Notes/Contents.html
* Python official documentation: https://docs.python.org/3/tutorial/index.html

Now that you have taken your first steps with Python, you are ready to move on to the next tutorial: ["A Quick Tour of Variables and Data Types in Python"](https://jovian.ml/aakashns/python-variables-and-data-types).

## Questions for Revision

1. What is a Jupyter notebook?
2. How do you add a new code cell below an existing cell?
3. How do you add a new Markdown cell below an existing cell?
4. How do you convert a code cell to a Markdown cell or vice versa?
5. How do you execute a code cell within Jupyter?
6. What the different arithmetic operations supported in Python?
7. How do you perform arithmetic operations using Python?
8. What is the difference between the `/` and the `//` operators?
9. What is the difference between the `*` and the `**` operators?
10. What is the order of precedence for arithmetic operators in Python?
11. How do you specify the order in which arithmetic operations are performed in an expression involving multiple operators?
12. How do you solve a multi-step arithmetic word problem using Python?
13. What are variables? Why are they useful?
14. How do you create a variable in Python?
15. What is the assignment operator in Python?
16. What are the rules for naming a variable in Python?
17. How do you view the value of a variable?
18. How do you store the result of an arithmetic expression in a variable?
19. What happens if you try to access a variable that has not been defined?
20. How do you display messages in Python?
21. What type of inputs can the print function accept?
22. What are code comments? How are they useful?
23. What are the different ways of creating comments in Python code?
24. What are the different comparison operations supported in Python?
25. What is the result of a comparison operation?
26. What is the difference between `=` and `==` in Python?
27. What are the logical operators supported in Python?
28. What is the difference between the `and` and `or` operators?
29. Can you use comparison and logical operators in the same expression?
30. What is the purpose of using parentheses in arithmetic or logical expressions?
31. What is Markdown? Why is it useful?
32. How do you create headings of different sizes using Markdown?
33. How do you create bulleted and numbered lists using Markdown?
34. How do you create bold or italic text using Markdown?
35. How do you include links & images within Markdown cells?
36. How do you include code blocks within Markdown cells?
37. Is it possible to execute the code blocks within Markdown cells?
38. How do you upload and share your Jupyter notebook online using Jovian?
39. What is the purpose of the API key requested by jovian.commit ? Where can you find the API key?
40. Where can you learn about arithmetic, conditional and logical operations in Python?


## Solution for Exercise



> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

In [54]:
plane_cost=5000
total_people_signed_up=29
ticket_cost=200

In [55]:
amount_received_by_company=total_people_signed_up*ticket_cost

In [56]:
total_profit=amount_received_by_company-plane_cost

In [57]:
total_profit

800

The travel company is in a profit of $800!

https://www.irurajackblogs.co.ke